hack/extract-licenses.go

```diff
@@ -0,0 +1,235 @@
+Go modules are downloaded to the Go module cache which can be changed via
+the environment: https://go.dev/ref/mod#module-cache`,
+		))
+	}
+	if _ = flags.Parse(os.Args[1:]); flags.NArg() < 2 || slices.ContainsFunc(
+		os.Args, func(arg string) bool { return arg == "-help" || arg == "--help" },
+	) {
+		flags.Usage()
+		os.Exit(2)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
+	go func() { <-signals; cancel() }()
+
+	// Create the target directory.
+	if err := os.MkdirAll(flags.Arg(0), 0o755); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+
+	// Extract module information from remaining arguments.
+	modules := identifyModules(ctx, flags.Args()[1:]...)
+
+	// Ignore packages from Crunchy Data. Most are not available in any [proxy],
+	// and we handle their licenses elsewhere.
+	//
+	// This is also a quick fix to avoid the [replace] directive in our projects.
+	// The logic below cannot handle them. Showing xxhash versus a replace:
+	//
+	//  dep	github.com/cespare/xxhash/v2	v2.3.0	h1:UL815xU9SqsFlibzuggzjXhog7bL6oX9BbNZnL2UFvs=
+	//  dep	github.com/crunchydata/postgres-operator	v0.0.0-00010101000000-000000000000
+	//  =>	./postgres-operator	(devel)
+	//
+	// [proxy]: https://go.dev/ref/mod#module-proxy
+	// [replace]: https://go.dev/ref/mod#go-mod-file-replace
+	modules = slices.DeleteFunc(modules, func(s string) bool {
+		return strings.HasPrefix(s, "github.com/crunchydata/")
+	})
+
+	// Download modules to the Go module cache.
+	directories := downloadModules(ctx, modules...)
+
+	// Gather license files from every module into the target directory.
+	for module, directory := range directories {
+		for _, license := range findLicenses(directory) {
+			relative := module + strings.TrimPrefix(license, directory)
+			destination := filepath.Join(flags.Arg(0), relative)
+
+			var data []byte
+			err := ctx.Err()
+
+			if err == nil {
+				err = os.MkdirAll(filepath.Dir(destination), 0o755)
+			}
+			if err == nil {
+				data, err = os.ReadFile(license)
+			}
+			if err == nil {
+				//nolint:gosec // gosec warns on permissions more open than 600
+				// but we need these licenses to be readable by all
+				err = os.WriteFile(destination, data, 0o644)
+			}
+			if err == nil {
+				//nolint:forbidigo // This is an intentional print to console to inform the user
+				fmt.Println(license, "=>", destination)
+			}
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				os.Exit(1)
+			}
+		}
+	}
+}
+
+func downloadModules(ctx context.Context, modules ...string) map[string]string {
+	var stdout bytes.Buffer
+
+	// Download modules and read their details into a series of JSON objects.
+	// - https://go.dev/ref/mod#go-mod-download
+	//nolint:gosec // Suppressing unnecessary warning re: potentially tainted inputs (G204)
+	cmd := exec.CommandContext(ctx, os.Getenv("GO"), append([]string{"mod", "download", "-json"}, modules...)...)
```

Comment:
⛏️ `gosec` has a [syntax](https://github.com/securego/gosec#annotating-code) for disabling specific rules inline that might work here. (untested)


```suggestion
	//gosec:disable G204 -- Use this environment variable to switch Go versions without touching PATH
	cmd := exec.CommandContext(ctx, os.Getenv("GO"), append([]string{"mod", "download", "-json"}, modules...)...)
```